controllers/sandbox_controller.go

```diff
@@ -537,6 +547,17 @@ func checkSandboxExpiry(sandbox *sandboxv1alpha1.Sandbox) (bool, time.Duration)
 	return false, requeueAfter
 }
 
+// sandboxMarkedExpired checks if the sandbox is already marked as expired
```

Comment:
Instead of manually iterating over conditions, consider using `meta.FindStatusCondition(sandbox.Status.Conditions, string(sandboxv1alpha1.SandboxConditionReady))` from `k8s.io/apimachinery/pkg/api/meta`. It simplifies the code and is the standard way to look up conditions.

Reply:
Done